### PR TITLE
Add Native Staking transaction data decoders

### PR DESCRIPTION
**Pull request by hectorgomezv**
## Summary
As part of the Native Staking transaction mapping implementation in the CGW, this PR adds _local_ decoding for these transactions. 

The point of this PR is to mitigate the risk of not having the Kiln contracts available (or the data-decoding endpoint unreachable) on the Safe Transaction Service (the main source for transaction data decoding). This PR provides a fallback for that scenario.

## Changes
- Adds `KilnDecoder` along with the Kiln Native Staking contract ABIs.
- Adds a fallback to `TransactionsViewService` that tries to decode the data in case the Transaction Service cannot.

### CODE REVIEW

File: src/domain/staking/contracts/decoders/kiln-decoder.helper.ts [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+    }
+    try {
+      const decoded = this.decodeFunctionData({ data });
+      if (decoded.functionName !== 'deposit') {
+        throw new Error('Data is not of deposit type');
+      }
+      return {
+        method: decoded.functionName,
+        parameters: [],
+      };
```
Comment (iamacook) [inline src/domain/staking/contracts/decoders/kiln-decoder.helper.ts:60]: Nit: each decoder is method specific so I don't think we need expose the `functionName`/can just return the parameters. What do you think?

File: src/domain/staking/contracts/decoders/kiln-decoder.helper.ts [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+] as const;
+
+export type KilnRequestValidatorsExitParameters = {
+  name: '_publicKeys';
+  type: 'bytes';
+  value: `0x${string}`;
+  valueDecoded: null;
+};
+export type KilnBatchWithdrawCLFeeParameters =
+  KilnRequestValidatorsExitParameters;
```
Comment (iamacook) [inline src/domain/staking/contracts/decoders/kiln-decoder.helper.ts:36]: I suggest we keep the decoded values as is and format it to match the backend response where it is used. What do you think?

File: src/routes/transactions/transactions-view.service.ts [hunk 3]
```diff
@@ -61,12 +63,16 @@ export class TransactionsViewService {
...
-        };
+        const { data } = args.transactionDataDto;
+        const fallbackDataDecoded =
+          this.kilnDecoder.decodeDeposit(data) ??
+          this.kilnDecoder.decodeValidatorsExit(data) ??
+          this.kilnDecoder.decodeBatchWithdrawCLFee(data);
+        if (!fallbackDataDecoded) {
+          throw new Error('Transaction data could not be decoded');
+        }
+        return fallbackDataDecoded;
```
Comment (iamacook) [inline src/routes/transactions/transactions-view.service.ts:75]: Do you think we should assign the fallback value here, or in the mappings? The reason being, `deposit` is a very generic signature and could be returned for non-Kiln interactions.

I suggest we also add a test case for "generic" signatures as well. What do you think?

Comment (hectorgomezv) [reply src/routes/transactions/transactions-view.service.ts]: I've moved the fallback decoding to the specific private function for each transaction type in [3fe322c](https://github.com/safe-global/safe-client-gateway/pull/1903/commits/3fe322ce083219a23b36488475b54d71191bc49b)

File: src/domain/staking/contracts/decoders/kiln-decoder.helper.ts (hunk 1 above)
Comment (hectorgomezv) [reply src/domain/staking/contracts/decoders/kiln-decoder.helper.ts:60]: After an offline sync, we decided to keep the `method` property in the return type. I've done so in [8fdaed7](https://github.com/safe-global/safe-client-gateway/pull/1903/commits/8fdaed7ac25cd5458e6c188bd282f238fd9ad3ba)

**Review by iamacook — APPROVED** [review]

**Pull request merged**